google-account-plugin/src/com/google/cloud/tools/intellij/stats/GoogleUsageTracker.java

```
@@ -36,64 +34,76 @@
-  @NonNls
-  private static final String ANALYTICS_APP = "Cloud Tools for IntelliJ";
-  @NonNls
-  private static final String INTELLIJ_EDITION = "intellij.edition";
+  private static final String ANALYTICS_APP = "gcloud-intellij";
 
   private final String analyticsId;
 
   public GoogleUsageTracker() {
     this.analyticsId = UsageTrackerManager.getInstance().getAnalyticsProperty();
   }
 
-  private static final List<? extends NameValuePair> analyticsBaseData = ImmutableList
-      .of(new BasicNameValuePair("v", "1"),
-          new BasicNameValuePair("t", "event"),
-          new BasicNameValuePair("an", ANALYTICS_APP),
-          new BasicNameValuePair("av", ApplicationInfo.getInstance().getFullVersion()),
-          new BasicNameValuePair("cid",
+  private static final List<BasicNameValuePair> analyticsBaseData = ImmutableList
+      .of(new BasicNameValuePair("v", "1"),         // Protocol version
+          new BasicNameValuePair("t", "pageview"),  // Note the "pageview" type, not "event"
+          new BasicNameValuePair("ni", "0"),        // Non-interactive? Report as interactive.
+          new BasicNameValuePair("cid",             // UUID for this IntelliJ client
               UpdateChecker.getInstallationUID(PropertiesComponent.getInstance())));
 
+  /**
+   * Send a (virtual) "pageview" ping to the Cloud-platform-wide Google Analytics Property.
+   *
+   * @param eventCategory Ignored for plugins installed on IntelliJ. Retained to maintain backward
+   *     compatibility on Android Studio.
+   * @param eventAction On IntelliJ, this is the primary event action.
+   * @param eventLabel On IntelliJ, reported as a metadata key.
+   * @param eventValue On IntelliJ, reported as a metadata value.
+   */
   @Override
   public void trackEvent(@NotNull String eventCategory,
       @NotNull String eventAction,
       @Nullable String eventLabel,
       @Nullable Integer eventValue) {
+    if (eventValue != null) {
```

Thats actually even harder to read to me. Just leave it how you had it now that I understand what you meant. Just to confirm: if eventValue is defined but eventLabel is null, do you really want to throw an exception?